DOC removed unnecessary use of parameter in example rename function call

KatemaNL opened the pull request:
removed unnecessary (and misleading) use of index parameter in docstring example function call of rename function

xref #26248 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Commit:
DOC removed unnecessary use of parameter in example rename function call

removed unnecessary (and misleading) use of index parameter in docstring example function call of rename function

mroeschke commented on pandas/core/frame.py:
```diff
@@ -3996,19 +3996,19 @@ def rename(self, *args, **kwargs):
         intent.
 
         >>> df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
-        >>> df.rename(index=str, columns={"A": "a", "B": "c"})
```
Instead of removing this line, I would change this example to show the impact of `index=str`:

```
In [4]: df.index
Out[4]: RangeIndex(start=0, stop=3, step=1)

In [5]: df.rename(index=str, columns={"A": "a", "B": "c"}).index
Out[5]: Index(['0', '1', '2'], dtype='object')
```

KatemaNL replied:
Great addition. I was wondering how to use the index parameter and this makes it perfectly clear.

Let's have two separate examples that show the use of either parameter isolated from the other.

```suggestion
Using columns parameter:
>>> df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
>>> df.rename(columns={"A": "a", "B": "c"})

Using index parameter:
>>> df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
>>> df.index
>>> df.rename(index=str).index
```

mroeschke replied:
Sure that works. I would change those example titles to `Remapping column names` and `Casting index labels to a different type`

KatemaNL replied:
Updated the suggested change.

KatemaNL replied:
Added a third example to be more clear:
```suggestion
Rename columns using a mapping:
>>> df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
>>> df.rename(columns={"A": "a", "B": "c"})

Rename rows using a mapping:
>>> df = pd.DataFrame(data={"A": [1, 2, 3], "B": [4, 5, 6]}, index=["X", "Y", "Z"])
>>> df.rename(index={"X": "x", "Y": "y", "Z": "z"})

Casting index labels to a different type:
>>> df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
>>> df.index
>>> df.rename(index=str).index
```

jreback replied:
@KatemaNL these all look good, ping when ready

KatemaNL replied:
@jreback Ready. How do we make sure the final suggested change finds its way and not the initial commit?

jreback replied:
only see 1
commit, did u push?

KatemaNL replied:
Just did, 2nd commit is available now

KatemaNL replied:
Fixed failed build checks. All checks have passed now.

Commit:
DOC added examples as discussed in the pull request review

added an example for using index parameter only
added an example for casting the index type

pep8speaks commented:
Hello @KatemaNL! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/core/frame.py`](https://github.com/pandas-dev/pandas/blob/6965f050cbb6f500687037ca699562704d3c694f/pandas/core/frame.py):

> [Line 4007:80](https://github.com/pandas-dev/pandas/blob/6965f050cbb6f500687037ca699562704d3c694f/pandas/core/frame.py#L4007): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)

Commit:
DOC fixed line to long

Commit:
DOC fixed doctest fail because of unexpected linebreak

mroeschke commented on pandas/core/frame.py:
```diff
@@ -3995,20 +3995,37 @@ def rename(self, *args, **kwargs):
         2  3  6
 
-        >>> df.rename(index=str, columns={"A": "a", "C": "c"})
+        Rename rows using a mapping:
```
I don't think this example is entirely necessarily, to demonstrate remapping twice.

KatemaNL replied:
This example is intended to demonstrate renaming the index/rows instead of the columns.

mroeschke replied:
I'd be okay with keeping this example if we remap the integer index label to string labels. So `df.rename(index={0: "x",  1: "y", 2: "z"})`

KatemaNL replied:
ok, nice twist

mroeschke replied:
The example assumes that you don't have to create the new dataframe below. In essence, all the examples can just use the first dataframe at the top.

KatemaNL replied:
New commit pushed.

@mroeschke, can you pay special attention to the correctness of the example with the errors parameter? Don't have the master branch version installed here.

mroeschke commented on pandas/core/frame.py:
```diff
@@ -3995,20 +3995,37 @@ def rename(self, *args, **kwargs):
+        z  3  6
+
+        Cast index labels to a different type:
+        >>> df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
```
Once the above example is removed, you don't need to construct the `df` twice in this example

KatemaNL replied:
example above won't be removed

KatemaNL replied:
I see what you mean now. Example is removed.

Commit:
DOC simplified examples

Commit:
DOC fixed the examples matching the output mentioned in the doc

mroeschke commented on pandas/core/frame.py:
```diff
@@ -3995,20 +3995,28 @@ def rename(self, *args, **kwargs):
-        0  1  4
-        1  2  5
-        2  3  6
+        Rename rows using a mapping:
```
Nit: can you replace `rows` with `index` here. Otherwise LGTM

KatemaNL replied:
done

Commit:
DOC terminology change

jreback commented:
lgtm. @mroeschke

mroeschke commented:
Thanks @KatemaNL
